ReactiveCocoa/NSObject+Lifetime.swift

```diff
@@ -1,14 +1,22 @@
 import Foundation
 import ReactiveSwift
 
-private var lifetimeKey: UInt8 = 0
-private var lifetimeTokenKey: UInt8 = 0
+extension Association {
+	/// Holds the `Lifetime` of the object.
+	fileprivate static let isSwizzled = AssociationKey<Bool>(contiguous + 4, default: false)
+
+	/// Holds the `Lifetime` of the object.
+	fileprivate static let lifetime = AssociationKey<Lifetime?>(contiguous + 4, default: nil)
+
+	/// Holds the `Lifetime.Token` of the object.
+	fileprivate static let lifetimeToken = AssociationKey<Lifetime.Token?>(contiguous + 5, default: nil)
```

Comment:
We can't have stored static property in generic types. So it wouldn't be a huge difference. It still needs to be backed by a top level constant.